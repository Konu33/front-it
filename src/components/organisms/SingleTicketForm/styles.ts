import {createStyles, makeStyles, Theme} from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    formWrapper: {
      display: 'flex',
      flexFlow: 'column nowrap',
      margin: theme.spacing(6, 'auto'),
      width: '100%',
      maxWidth: 630,
      height: '100%',
    },
    formHeader: {
      alignSelf: 'center',
      display: 'flex',
      flexFlow: 'column nowrap',
      alignItems: 'center',
    },
    form: {
      minWidth: 250,
    },
    securityIcon: {
      backgroundColor: theme.palette.secondary.dark,
      margin: theme.spacing(1),
    },
    formField: {
      display: 'block',
      margin: theme.spacing(2, 0),
    },
    title: {
      marginRight: 'auto',
      marginBottom: '10px',
    },
    textField_root: {
      '& input:-webkit-autofill': {
        '-webkit-text-fill-color': theme.palette.text.primary,
        transition: 'backgrounds-color 5000s ease-in-out 0s',
        '&:hover, &:focus, &:active': {
          '-webkit-text-fill-color': theme.palette.text.primary,
          transition: 'backgrounds-color 5000s ease-in-out 0s',
        },
      },
    },
    submitBtn: {
      width: '150px',
      marginLeft: 'auto',
      color: theme.palette.background.default,
    },
    submitBtn_label: {
      color: 'inherit',
    },
    errorSnackbar: {
      width: '100%',
    },
  })
);

export default useStyles;
