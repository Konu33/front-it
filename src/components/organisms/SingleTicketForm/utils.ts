import * as yup from 'yup';

export interface IFormValues {
  description: string;
}

const initialValues = {description: ''};

const validationSchema = yup.object().required().shape({
  description: yup.string().required(),
});

export {initialValues, validationSchema};
