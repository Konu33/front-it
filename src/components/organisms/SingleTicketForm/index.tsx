import TextField from '@material-ui/core/TextField';
import classNames from 'classnames';
import {Formik} from 'formik';
import {FormikHelpers, FormikProps} from 'formik/dist/types';
import React, {memo, useRef} from 'react';

import useMergeClasses from '../../../hooks/useMergeClasses';
import IComment from '../../../models/IComment';
import ITicket from '../../../models/ITicket';
import {OverrideClassesProp} from '../../../utils/styling-utils';
import ProgressButton from '../../atoms/ProgressButton';
import CommentsList from '../../molecules/CommentsList';
import useStyles from './styles';
import * as utils from './utils';

export interface ISingleTicketFormFormProps {
  onSubmit: (values: typeof utils.initialValues, helpers: FormikHelpers<typeof utils.initialValues>) => Promise<void>;
  classes?: OverrideClassesProp<typeof useStyles>;
  comments: IComment[];
  ticket: ITicket;
  removeComment: (commentId: number) => void;
}

function SingleTicketForm(props: ISingleTicketFormFormProps) {
  let {onSubmit: onFormSubmit, removeComment} = props;
  const formikRef = useRef<FormikProps<typeof utils.initialValues>>(null);
  const classes = useMergeClasses(useStyles(props), props.classes);
  const {comments, ticket} = props;

  return (
    <div className={classes.formWrapper}>
      <CommentsList comments={comments} ticket={ticket} removeComment={removeComment} />
      <Formik
        initialValues={utils.initialValues}
        validationSchema={utils.validationSchema}
        onSubmit={onFormSubmit}
        innerRef={formikRef}
      >
        {({values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting}) => (
          <form onSubmit={handleSubmit} className={classes?.form} data-testid='comments'>
            <TextField
              inputProps={{'data-testid': 'description'}}
              name={'description'}
              label={'Description'}
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.description}
              variant={'outlined'}
              error={Boolean(touched.description && errors.description)}
              className={classes?.formField}
              multiline
              rows={4}
              fullWidth
              disabled={isSubmitting}
            />

            <ProgressButton
              data-testid='button'
              type={'submit'}
              color={'primary'}
              variant={'contained'}
              inProgress={isSubmitting}
              className={classNames(classes?.formField, classes?.submitBtn)}
              ButtonProps={{
                classes: {
                  text: classes?.submitBtn_label,
                },
              }}
            >
              Add Comment
            </ProgressButton>
          </form>
        )}
      </Formik>
    </div>
  );
}

export default memo(SingleTicketForm);
