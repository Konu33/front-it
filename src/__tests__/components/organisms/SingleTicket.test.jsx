import {cleanup, fireEvent, render, waitFor} from '@testing-library/react';
import React from 'react';

import SingleTicketForm from '../../../components/organisms/SingleTicketForm';

afterEach(cleanup);
const ticket = {
  id: 2,
  title: 'testtetowy',
  createdBy: 1,
  description: 'sdsdxxxxxxxxxxxxxxxx',
  addDate: '11-01-2021 19:31:07',
  status: 1,
  assignedTo: 3,
};
const comments = [
  {
    id: 1,
    ticketId: 1,
    content: 'test contetn',
    addDate: '21-01-2021 18:00:11',
    createdBy: {
      name: 'Jan',
      surname: 'Kowalski',
      iconColor: '#03A9F4',
    },
  },
  {
    id: 2,
    ticketId: 1,
    content: 'test',
    addDate: '21-01-2021 18:03:11',
    createdBy: {
      name: 'Piotr',
      surname: 'Nowak',
      iconColor: '#673AB7',
    },
  },
];

test('should render SingleTicketForm component', async () => {
  const {getByTestId} = render(<SingleTicketForm ticket={ticket} />);
  expect(getByTestId('comments')).toBeInTheDocument();
});

test('should render SingleTicketForm form', async () => {
  const {getByTestId} = render(<SingleTicketForm ticket={ticket} comments={comments} />);
  const input = getByTestId('description');
  expect(input).toBeInTheDocument();
  expect(getByTestId('button')).toBeInTheDocument();
});
test('should change descrpiton field on change', async () => {
  const {getByTestId, asFragment} = render(<SingleTicketForm ticket={ticket} comments={comments} />);
  const input = getByTestId('description');

  fireEvent.change(input, {
    target: {value: 'new content'},
  });

  let inputValue;
  await waitFor(() => {
    inputValue = input.value;
  });

  expect(inputValue).toBe('new content');
  expect(asFragment()).toMatchSnapshot();
});
