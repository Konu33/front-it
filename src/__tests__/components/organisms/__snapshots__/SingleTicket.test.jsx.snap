// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change descrpiton field on change 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-formWrapper-53"
  >
    <div
      class="makeStyles-formHeader-72"
    >
      <div
        class="makeStyles-iconBox-66"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-iconSize-65"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </div>
       
      <h5
        class="MuiTypography-root makeStyles-title-71 MuiTypography-h5"
        data-testid="title"
      >
        testtetowy
      </h5>
       
      <div
        data-testid="ticket-desc"
      >
        sdsdxxxxxxxxxxxxxxxx
      </div>
    </div>
    <div
      class="makeStyles-comments-63"
    >
      <h5
        class="MuiTypography-root makeStyles-title-71 MuiTypography-h5"
        data-testid="list-title"
      >
        Comments
      </h5>
      <div
        class="makeStyles-commentList-69"
        data-testid="comments"
      >
        <div
          class="makeStyles-commentAuthorLeft-70"
        >
           Jan Kowalski
        </div>
         
        <div
          class="makeStyles-commentDescriptionLeft-64"
        >
          test contetn
        </div>
      </div>
      <div
        class="makeStyles-commentList-69"
        data-testid="comments"
      >
        <div
          class="makeStyles-commentAuthorRight-68"
        >
           Piotr Nowak
        </div>
         
        <div
          class="makeStyles-commentDescriptionRight-67"
        >
          test
        </div>
      </div>
    </div>
    <form
      class="makeStyles-form-55"
      data-testid="comments"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-formField-57 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Description
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            data-testid="description"
            name="description"
            rows="4"
          >
            new content
          </textarea>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-73 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-75 PrivateNotchedOutline-legendNotched-76"
            >
              <span>
                Description
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <span
        class="makeStyles-root-77 makeStyles-formField-57 makeStyles-submitBtn-60"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="button"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Add Comment
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </span>
    </form>
  </div>
</DocumentFragment>
`;
